addons/point_of_sale/models/pos_config.py

```diff
@@ -423,7 +423,8 @@ def write(self, vals):
                     if key == 'use_pricelist' and vals[key]:
                         continue
                     if key == 'available_pricelist_ids':
-                        removed_pricelist = set(self.available_pricelist_ids.ids) - set(vals[key][0][2])
+                        new_pricelist_ids = vals[key][0][2] if vals[key] and len(vals[key][0]) >= 3 else []
+                        removed_pricelist = set(self.available_pricelist_ids.ids) - set(new_pricelist_ids)
```

Comment:
We'll need a general fix for this because this issue can be observed from other x2many fields.

Reply:
Another example of similar issue is the field for multiple splash screens introduced in the kiosk feature. If you start a db with demo data, and try to delete one or two of the existing splash screens, saving the config won't make the change you requested.

Reply:
The multiple splash screens config doesn't seem to work for me, the button allows me to pick an image but nothing happens then. This fix is simple and just removes the Python error.

Reply:
I changed the fix to make the code work as I think it was expected to behave.